src/lib.rs

```diff
@@ -115,4 +122,140 @@
 
 mod promise;
 
-pub use promise::{Promise, Sender};
+pub use promise::{Promise, Sender, TaskType};
+
+#[cfg(feature = "smol")]
+static EXECUTOR: smol::Executor<'static> = smol::Executor::new();
+#[cfg(feature = "smol")]
+thread_local! {
+    static LOCAL_EXECUTOR: smol::LocalExecutor<'static> = smol::LocalExecutor::new();
+}
+
+/// 'Tick' the smol thread executor.
+///
+/// Poll promise will call this for you when using [`Promise::block_until_ready`] and friends.
+/// If so desired [`Promise::poll`] will run this for you with the `tick-poll` feature.
+#[cfg(feature = "smol")]
+pub fn tick() -> bool {
+    crate::EXECUTOR.try_tick()
+}
+
+/// 'Tick' the smol local thread executor.
+///
+/// Poll promise will call this for you when using [`Promise::block_until_ready`] and friends.
+/// If so desired [`Promise::poll`] will run this for you with the `tick-poll` feature.
+#[cfg(feature = "smol")]
+pub fn tick_local() -> bool {
+    crate::LOCAL_EXECUTOR.with(|exec| exec.try_tick())
+}
+
+#[cfg(test)]
+mod test {
+    use crate::Promise;
+
+    #[test]
+    fn it_spawns_threads() {
+        let promise = Promise::spawn_thread("test", || {
+            std::thread::sleep(std::time::Duration::from_secs(1));
+            0
+        });
+
+        assert_eq!(0, promise.block_and_take());
+    }
+
+    #[test]
+    #[cfg(feature = "smol")]
+    fn it_runs_async_threaded() {
+        let promise = Promise::spawn_async(async move { 0 });
+
+        assert_eq!(0, promise.block_and_take());
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    #[cfg(feature = "tokio")]
+    async fn it_runs_async_threaded() {
+        let promise = Promise::spawn_async(async move { 0 });
+
+        assert_eq!(0, promise.block_and_take());
+    }
+
+    #[test]
+    #[cfg(feature = "smol")]
+    fn it_runs_locally() {
+        let promise = Promise::spawn_local(async move { 0 });
+
+        assert_eq!(0, promise.block_and_take());
+    }
+
+    #[test]
+    #[cfg(feature = "smol")]
+    fn it_runs_background() {
+        let promise = Promise::spawn_async(async move {
+            let mut e = 0;
+            for i in -10000..0 {
+                e += i;
+            }
+            e
+        });
+        crate::tick();
```

Comment:
I guess this `tick()` isn't needed with the `tick-poll` feature?

Reply:
Oop my bad! You're right, it isn't needed.